js/tests/enhancedDate.spec.js

```diff
@@ -0,0 +1,69 @@
+// describing spec (what's being tested)
+describe("enhancedDate", function () {
+
+	// Test each method within object
+
+	// setDate
+	describe("setDate", function () {
```

Comment:
One more obvious test is to simply test that the return value is of the expected type:
`expect(enhancedDate.setDate()).is.an.instanceof('Date');`